tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/EventProcessing/IssueProcessing.cs

```diff
@@ -41,45 +42,210 @@ public static async Task ProcessIssueEvent(GitHubEventClient gitHubEventClient,
+        //                                                                    labelSuggestions);
+        //                        }
+        //                        // At this point, it cannot be determined who this issue belongs to. Add
+        //                        // the needs-team-triage label instead of the needs-team-attention label
+        //                        else
+        //                        {
+        //                            gitHubEventClient.AddLabel(TriageLabelConstants.NeedsTeamTriage);
+        //                            addNeedsTeamAttention = false;
+        //                        }
+        //                    }
+
+        //                    // The needs-team-attention label is only added when it can be determined
+        //                    // who this issue belongs to.
+        //                    if (addNeedsTeamAttention)
+        //                    {
+        //                        gitHubEventClient.AddLabel(TriageLabelConstants.NeedsTeamAttention);
+        //                    }
+        //                    gitHubEventClient.AddLabel(TriageLabelConstants.NeedsTeamTriage);
+
+        //                }
+        //                // If there are no labels predicted add NeedsTriage to the issue
+        //                else
+        //                {
+        //                    gitHubEventClient.AddLabel(TriageLabelConstants.NeedsTriage);
+        //                }
+
+        //                // If the user is not a member of the Azure Org AND the user does not have write or admin collaborator permission.
+        //                // This piece is executed for every issue created that doesn't have labels or owners on it at the time of creation.
+        //                bool isMemberOfOrg = await gitHubEventClient.IsUserMemberOfOrg(OrgConstants.Azure, issueEventPayload.Issue.User.Login);
+        //                if (!isMemberOfOrg)
+        //                {
+        //                    bool hasAdminOrWritePermission = await gitHubEventClient.DoesUserHaveAdminOrWritePermission(issueEventPayload.Repository.Id, issueEventPayload.Issue.User.Login);
+        //                    if (!hasAdminOrWritePermission)
+        //                    {
+        //                        gitHubEventClient.AddLabel(TriageLabelConstants.CustomerReported);
+        //                        gitHubEventClient.AddLabel(TriageLabelConstants.Question);
+        //                    }
+        //                }
+        //            }
+        //        }
+        //    }
+        //}
+        // TODO-END
+
+        // This is the IntitialIssueTriage as it exists today. It's the old rule that needs to be replaced by the commented out
+        // function above.
         public static async Task InitialIssueTriage(GitHubEventClient gitHubEventClient, IssueEventGitHubPayload issueEventPayload)
```

Comment:
I think it would be helpful to have both the manual and automated triage rules exist side-by-side so that individual repositories can opt-out of automated triage if needed.

Reply:
@jsquire if needed we can do this but only if it's needed. The updates required to add a completely new rule require not only the auto-PRs for each repository required from updating the version but also individual PRs to add the rule to the config file. Being that the teams committed to having the AzureSdkOwners in their CODEOWNERS files, I wouldn't want to do this unless it was absolutely necessary.